### PR TITLE
Add 3.4.0 release note and news and update links

### PR DESCRIPTION
**Pull request by xinrong-meng**
Add 3.4.0 release note and news and update links

### CODE REVIEW

File: releases/_posts/2023-04-13-spark-release-3-4-0.md [hunk 1]
```diff
@@ -0,0 +1,420 @@
...
+* Relax ordering constraint for CREATE TABLE column options ([SPARK-40944](https://issues.apache.org/jira/browse/SPARK-40944))
+* SQL Equivalent for Dataframe overwrite command ([SPARK-40956](https://issues.apache.org/jira/browse/SPARK-40956))
+* Support Generate with no required child output to host outer references ([SPARK-41441](https://issues.apache.org/jira/browse/SPARK-41441))
+* ORDER BY ALL ([SPARK-41637](https://issues.apache.org/jira/browse/SPARK-41637))
+* GROUP BY ALL ([SPARK-41635](https://issues.apache.org/jira/browse/SPARK-41635))
+* Add flatMapSortedGroups and cogroupSorted ([SPARK-38591](https://issues.apache.org/jira/browse/SPARK-38591))
+* Support subqueries with correlated non-equality predicates ([SPARK-36114](https://issues.apache.org/jira/browse/SPARK-36114))
+* Support subqueries with correlation through UNION/INTERSECT/EXCEPT ([SPARK-36124](https://issues.apache.org/jira/browse/SPARK-36124))
+
+#### ANSI
```
Comment (gengliangwang) [inline releases/_posts/2023-04-13-spark-release-3-4-0.md]:
```suggestion
#### ANSI Compliance
```

**Review by dongjoon-hyun — COMMENTED** [review]
Also cc @sunchao

File: releases/_posts/2023-04-13-spark-release-3-4-0.md [hunk 2]
```diff
@@ -0,0 +1,420 @@
...
+
+* ANSI mode: always return null on invalid access to map column ([SPARK-40066](https://issues.apache.org/jira/browse/SPARK-40066))
```
Comment (gengliangwang) [inline releases/_posts/2023-04-13-spark-release-3-4-0.md]:
```suggestion
* ANSI SQL mode: always return null on invalid access to map column ([SPARK-40066](https://issues.apache.org/jira/browse/SPARK-40066))
```

**Review by dongjoon-hyun — COMMENTED** [review]
@xinrong-meng , you need to build `html`. Please refer `README.md`.

File: releases/_posts/2023-04-13-spark-release-3-4-0.md [hunk 3]
```diff
@@ -0,0 +1,420 @@
...
+* Support cast of decimals to ANSI intervals ([SPARK-40014](https://issues.apache.org/jira/browse/SPARK-40014))
+* Return wider ANSI interval types from the percentile functions ([SPARK-40151](https://issues.apache.org/jira/browse/SPARK-40151))
+* Support cast of ANSI intervals to decimals ([SPARK-39470](https://issues.apache.org/jira/browse/SPARK-39470))
+* Support casting intervals to integrals in ANSI mode ([SPARK-39451](https://issues.apache.org/jira/browse/SPARK-39451))
+* Harden SQLSTATE usage for error classes ([SPARK-41994](https://issues.apache.org/jira/browse/SPARK-41994))
+
+#### Functions
+
+* Support table-valued generator functions in the FROM clause ([SPARK-41594](https://issues.apache.org/jira/browse/SPARK-41594))
+* Support ANSI Aggregate Function: REGR_SXY ([SPARK-37681](https://issues.apache.org/jira/browse/SPARK-37681))
```
Comment (gengliangwang) [inline releases/_posts/2023-04-13-spark-release-3-4-0.md:87]: Shall we group all the `ANSI Aggregate Function` together?

File: releases/_posts/2023-04-13-spark-release-3-4-0.md [hunk 4]
```diff
@@ -0,0 +1,420 @@
...
+* Introducing a streaming checkpoint file manager based on Hadoop's Abortable interface ([SPARK-40039](https://issues.apache.org/jira/browse/SPARK-40039))
+* Deprecate Trigger.Once and Promote Trigger.AvailableNow ([SPARK-39805](https://issues.apache.org/jira/browse/SPARK-39805))
+* Expose the information of catalog table to the logical plan in streaming query ([SPARK-39564](https://issues.apache.org/jira/browse/SPARK-39564))
+* Support collecting metrics from streaming sinks ([SPARK-38564](https://issues.apache.org/jira/browse/SPARK-38564))
+* Deprecate DStream API ([SPARK-42075](https://issues.apache.org/jira/browse/SPARK-42075))
+* Flip the default value of Kafka offset fetching config ([SPARK-40844](https://issues.apache.org/jira/browse/SPARK-40844))
+* Provide cloned spark session in DataFrame in user function for foreachBatch sink in PySpark ([SPARK-41379](https://issues.apache.org/jira/browse/SPARK-41379))
+
+
+### Spark Connect
```
Comment (gengliangwang) [inline releases/_posts/2023-04-13-spark-release-3-4-0.md:280]: No features for Spark Connect?

File: releases/_posts/2023-04-13-spark-release-3-4-0.md (hunk 1 above)
Comment (xinrong-meng) [reply releases/_posts/2023-04-13-spark-release-3-4-0.md]: Updated.

File: releases/_posts/2023-04-13-spark-release-3-4-0.md (hunk 2 above)
Comment (xinrong-meng) [reply releases/_posts/2023-04-13-spark-release-3-4-0.md]: Updated.

File: releases/_posts/2023-04-13-spark-release-3-4-0.md (hunk 3 above)
Comment (xinrong-meng) [reply releases/_posts/2023-04-13-spark-release-3-4-0.md:87]: Updated.

File: releases/_posts/2023-04-13-spark-release-3-4-0.md [hunk 5]
```diff
@@ -0,0 +1,420 @@
...
+* GROUP BY ALL ([SPARK-41635](https://issues.apache.org/jira/browse/SPARK-41635))
+* Add flatMapSortedGroups and cogroupSorted ([SPARK-38591](https://issues.apache.org/jira/browse/SPARK-38591))
+* Support subqueries with correlated non-equality predicates ([SPARK-36114](https://issues.apache.org/jira/browse/SPARK-36114))
+* Support subqueries with correlation through UNION/INTERSECT/EXCEPT ([SPARK-36124](https://issues.apache.org/jira/browse/SPARK-36124))
+
+#### ANSI Compliance
+
+* ANSI SQL mode: always return null on invalid access to map column ([SPARK-40066](https://issues.apache.org/jira/browse/SPARK-40066))
+* Support double quoted identifiers ([SPARK-40585](https://issues.apache.org/jira/browse/SPARK-40585))
+* ANSI SQL mode: Round/Bround should return an error on integer overflow [SPARK-42045](https://issues.apache.org/jira/browse/SPARK-42045))
```
Comment (ueshin) [inline releases/_posts/2023-04-13-spark-release-3-4-0.md]: nit: The left paren is missing before `[SPARK-`.

Comment (xinrong-meng) [reply releases/_posts/2023-04-13-spark-release-3-4-0.md]: Updated.

**Comment by xinrong-meng** [comment]
Also cc @HyukjinKwon @HeartSaVioR @cloud-fan @xuanyuanking @hvanhovell

**Comment by xinrong-meng** [comment]
Thanks @dongjoon-hyun! Let me build `html`s after resolving existing comments.

**Review by HyukjinKwon — APPROVED** [review]

**Comment by viirya** [comment]
A failure in CI:

```
Error during document generation:
             Error: YAML Exception reading /__w/spark-website/spark-website/news/_posts/2023-04-13-spark-3-4-0-released.md: (<unknown>): mapping values are not allowed in this context at line 6 column 7
```

**Comment by dongjoon-hyun** [comment]
Thank you for fixing it.

**Review by dongjoon-hyun — CHANGES_REQUESTED** [review]
Sorry but this PR looks wrong to me.

There exists too many files like the following. Apache Spark 3.4.0 news PR should not add this kind of files.
```
spark-website:3.4.0_doc_cont $ git diff asf-site --stat site/_site/docs/preview/api/scala/org/apache/spark/ml/clustering/GaussianMixture.html
 site/_site/docs/preview/api/scala/org/apache/spark/ml/clustering/GaussianMixture.html | 1931 +++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
 1 file changed, 1931 insertions(+)
```

@xinrong-meng . Please check the previous news commit. This PR is touching old all documents.

**Comment by HyukjinKwon** [comment]
I fixed and force-pushed it.

**Review by MaxGekk — APPROVED** [review]

**Review by yaooqinn — APPROVED** [review]

**Review by gengliangwang — APPROVED** [review]

**Comment by HyukjinKwon** [comment]
Thanks all.

Merged to asf-site branch.

**Comment by bjornjorgensen** [comment]
We must remove "This will become a table of contents (this text will be scraped)." from https://spark.apache.org/releases/spark-release-3-4-0.html

**Comment by HyukjinKwon** [comment]
@bjornjorgensen are you interested in making a PR please?

**Comment by bjornjorgensen** [comment]
@HyukjinKwon I have some problems with 
`
bundle install
Bundler 2.4.1 is running, but your lockfile was generated with 2.3.7. Installing Bundler 2.3.7 and restarting using that version.
Fetching gem metadata from https://rubygems.org/.
`
install failed!

**Comment by bjornjorgensen** [comment]
@HyukjinKwon If you have a working install of Bundler 
this is the line https://github.com/apache/spark-website/blob/74099512789a0f4099716b5e6cf0882eed6f5d13/releases/_posts/2023-04-13-spark-release-3-4-0.md?plain=1#L20

**Comment by HyukjinKwon** [comment]
uhoh, sure let me make a quick fix then.

**Comment by HyukjinKwon** [comment]
https://github.com/apache/spark-website/pull/449

**Comment by xinrong-meng** [comment]
Thank you all! Thanks @HyukjinKwon for the fix.

**Comment by dongjoon-hyun** [comment]
+1, late LGTM (for this and followup). Thank you, all!

**Pull request closed without merging**